src/Cnab.Bundle/Parameter.cs

```diff
@@ -6,19 +7,41 @@
 namespace Cnab.Bundle
 {
     [JsonConverter(typeof(ParameterConverter))]
-    public abstract class Parameter
+    public interface IParameter
+    {
+        string DataType { get; set; }
+        bool Required { get; set; }
+        ParameterMetadata Metadata { get; set; }
+        Location Destination { get; set; }
+
+        bool IsValid(string value);
+        bool IsValid(int value);
+        bool IsValid(bool value);
```

Comment:
The current alternative to this would be using `object` as type for the parameter, thing I strongly dislike.